src/highlight/SingleHighlightAnnotation.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import classNames from 'classnames';
+import HighlightAnnotation from './HighlightAnnotation';
+import HighlightCanvas from './HighlightCanvas';
+import HighlightSvg from './HighlightSvg';
+import { AnnotationHighlight } from '../@types';
+
+type Props = {
+    annotation: Pick<AnnotationHighlight, 'id' | 'target'>;
+    className?: string;
+};
+
+const SingleHighlightAnnotation = (
+    { annotation, className }: Props,
+    ref: React.Ref<HTMLAnchorElement>,
+): JSX.Element => {
+    const {
+        id,
+        target: { shapes },
+    } = annotation;
+    return (
+        <div className={classNames('ba-SingleHighlightAnnotation', className)}>
+            <HighlightCanvas annotations={[annotation]} />
```

Comment:
This will create a new array every time it is rendered, which isn't ideal. It seems like there's some duplication between the HighlightList and this component. Can we pull out the reused portions to simplify the component structure?